api/test/nostd/variant_test.cc

```diff
@@ -48,7 +48,7 @@ TEST(VariantTest, Get)
   EXPECT_EQ(nostd::get<int>(w), 12);
   EXPECT_EQ(*nostd::get_if<int>(&v), 12);
   EXPECT_EQ(nostd::get_if<float>(&v), nullptr);
-#if __EXCEPTIONS
+#if __EXCEPTIONS || (defined(OPENTELEMETRY_STL_VERSION) && (OPENTELEMETRY_STL_VERSION >= 2020))
```

Comment:
std::variant is introduced in C++17, not C++20.

Please investigate (local CXX 17 MSVC build) if the logic should be `OPENTELEMETRY_STL_VERSION >= 2017` instead.

Also, somehow the existing code passed in C++14, C++17, C++20 builds in CI with GCC.

Not sure which platform and/or which compiler do set __EXCEPTIONS, please clarify if you know the context.

Reply:
Found out that __EXCEPTIONS is set by GCC and Clang, except when building with -fno-exceptions.

This is likely why a GCC build with exceptions passed this test.

Reply:
Thanks for looking into it. Will try MSVC C++ 17 build.